### ISSUE TITLE
Distinction between first and last name

### ISSUE THREAD

**Issue by aspdye** [owncloud/contacts#324]

**Comment by jancborchardt** [comment]
We should definitely try to do this in the backend. It’s already painful to need to put in the address in 5 or 6 different fields (everyone _HATES_ forms) so we definitely should not have separate input fields for the name. The old Contacts app didn’t have it either, at least not by default.

cc @tanghus @jbtbnl @babelouest if you feel like contributing again. :)

**Comment by DeepDiver1975** [comment]
Name editing in the old contacts app

![bildschirmfoto von 2016-03-31 11-43-27](https://cloud.githubusercontent.com/assets/1005065/14171922/d1e8b346-f735-11e5-9f84-e0dcb98e3480.png)

**Comment by DeepDiver1975** [comment]
editing addresses in old contacts app

![bildschirmfoto von 2016-03-31 11-44-31](https://cloud.githubusercontent.com/assets/1005065/14171948/fc19a7ce-f735-11e5-8ce2-dcf14067923d.png)

**Comment by jancborchardt** [comment]
Damn … then it was probably wishful thinking, or it of course works like that on Android etc. ;)

**Comment by skjnldsv** [comment]
In Android I think the contact app split automatically the Name you enter and divide the data into the FN field.

Example, if you type "John Doe", it will set John as first name and Doe as last name.
Maybe we should do that with a div expand possibility to correct the FN fields?

**Comment by ionum** [comment]
Currently new Contact only have FN-Value (Displayname) and N-Value (Full Name with Family Name, Given Name, Additional Names, Honorific Prefixes, and Honorific Suffixes as in RFC 2426) is not editable.

**Comment by ionum** [comment]
A lot of progams use algorithms to split names (eg. Outlook, Android, iOS), but none works perfectly. there are too much ways to write a full name....

**Comment by aspdye** [comment]
#265

**Comment by nWRqenkF** [comment]
related to #276 ??

**Comment by nWRqenkF** [comment]
@ionum Somehow we already have an algorithm: "do nothing". Which is really annoying when using any other contact application.
Personally I would prefer the 8.1 contacts behaviour:
- Show only a single input field and try to split the input to first name and given name with an algorithm
- Give the user the possibility to display and edit the name detail fields, if he needs to correct the result of the algorithm

**Comment by XZS1cpXn** [comment]
One of the reasons I didn't upgrade to OwnCloud 9 yet is because of contacts app. Before if I had a contact named for example, "John Something Anything Doe" and I just wanted to display as "John Doe" but still keep the first name as "John Something" and last name "Anything Doe" I could, now my contact keeps displaying the full name, which is really inconvenient. I was very surprised that a simple feature like this disappeared with the update, things should get better with the updates not worse :)

**Comment by jancborchardt** [comment]
@XZS1cpXn yes, sure. Just for clarification, the reason for this is that it's a completely new app which replaced the old one. Since unfortunately resources are scarce, some things are not done yet. :)

**Comment by gazbarg** [comment]
Moreover, i think it could help with contact sorting (issue #208).

**Comment by Rello** [comment]
my first try #418

**Comment by Henni** [comment]
might be relevant: https://github.com/joshfraser/JavaScript-Name-Parser

**Comment by DeepDiver1975** [comment]
That libs dead Baby, that libs dead!
Last commit 2 years old

**Comment by skjnldsv** [comment]
:skull: Nooooooooo

**Comment by phsc84** [comment]
Shouldn't this issue be fixed with #437?

**Comment by jancborchardt** [comment]
@phsc yup, thanks!

We can do enhancements in the future, but for now this should fix it.

**Comment by jancborchardt** [comment]
Or wait – this was for the other issue.

This issue is about parsing the name, which is an entirely different beast. ;)
